test/functional/rpc_scanblocks.py

```diff
@@ -71,6 +79,31 @@ def run_test(self):
         assert(blockhash in node.scanblocks(
             "start", [{"desc": f"pkh({parent_key}/*)", "range": [0, 100]}], height)['relevant_blocks'])
 
+        # check that false-positives are included in the result now; note that
+        # finding a false-positive at runtime would take too long, hence we simply
+        # use a pre-calculated one that collides with the regtest genesis block's
+        # coinbase output and verify that their BIP158 ranged hashes match
+        genesis_blockhash = node.getblockhash(0)
+        genesis_spks = bip158_relevant_scriptpubkeys(node, genesis_blockhash)
+        assert_equal(len(genesis_spks), 1)
+        genesis_coinbase_spk = list(genesis_spks)[0]
+        false_positive_spk = bytes.fromhex("001400000000000000000000000000000000000cadcb")
+
+        genesis_coinbase_hash = bip158_basic_element_hash(genesis_coinbase_spk, 1, genesis_blockhash)
+        false_positive_hash = bip158_basic_element_hash(false_positive_spk, 1, genesis_blockhash)
+        assert_equal(genesis_coinbase_hash, false_positive_hash)
+
+        assert(genesis_blockhash in node.scanblocks(
+            "start", [{"desc": f"raw({genesis_coinbase_spk.hex()})"}], 0, 0)['relevant_blocks'])
+        assert(genesis_blockhash in node.scanblocks(
+            "start", [{"desc": f"raw({false_positive_spk.hex()})"}], 0, 0)['relevant_blocks'])
+
+        # check that the filter_false_positives option works
```

Comment:
Should we also include a test that breaks if the block containing the false positive gets pruned?

Reply:
Considering that false positives can only be generated in the genesis block and that the genesis block cannot be pruned, I'm not sure how to test this. I will leave this test as a follow-up for a brave soul.

Reply:
could hand-craft the first block at the python test suite side.

Reply:
My assumptions were wrong, I thought the genesis block could not get pruned. Added a test case for this https://github.com/bitcoin/bitcoin/compare/fd0ed48ecd9f0d65acc3199a50b3ca6479aa5a3e..c39234b070a0d2102af835fda77f3a78d4a631d7